packages/data-transport-layer/src/db/transport-db.ts

```diff
@@ -379,15 +379,38 @@ export class TransportDB {
     if (index === null) {
       return null
     }
-
-    return this.db.get<TEntry>(`${key}:index`, index)
+    let entry = await this.db.get<TEntry>(`${key}:index`, index)
+    entry = stringify(entry)
+    return entry
   }
 
   private async _getEntries<TEntry extends Indexed>(
     key: string,
     startIndex: number,
     endIndex: number
   ): Promise<TEntry[] | []> {
-    return this.db.range<TEntry>(`${key}:index`, startIndex, endIndex)
+    const entries = await this.db.range<TEntry>(
+      `${key}:index`,
+      startIndex,
+      endIndex
+    )
+    const results = []
+    for (const entry of entries) {
+      results.push(stringify(entry))
+    }
+    return results
+  }
+}
+
+function stringify(entry) {
```

Comment:
This could probably be made cleaner but we're planning to do a major overhaul of this stuff anyway so this is fine IMO.